NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion
# JA-viewAllPosts-3

# Description
Given the user is in the Tabloid application
When they select the Posts menu option
Then they should be directed to the Posts list page
And each post in the list should display the title, author and category
And the list should ONLY contain approved Posts
And the list should ONLY contain Posts with a publication date that is in the past
And the list should be in order of publication date with the most recent on top

**Related Ticket(s)**
Ticket#3 View All Posts -https://trello.com/c/V0r5NT3H/3-3-view-all-posts

**Type of Change**

# Testing Instructions

#Setup (can be skipped if you already have the Tabloid database configured with provided seed data)
- Clone this project’s repository to your machine
  - If you’ve already done this and are currently working on a different branch of the project, be sure to `git add` and `git commit` whatever you’re working on on your branch, before you attempt to switch branches
  - Then `git checkout main` to switch to the project’s main branch
- Navigate to the root directory for this repo (yourParentFolderName/tabloidfullstack-the-body-slam-batallion)
- `git fetch -a` to retrieve the current remote-tracked branches for the project, you should see a branch named `JA-viewAllPosts-3` as one of the available branches
- `git checkout `JA-viewAllPosts-3` to switch to my branch
- If you don't see the most recent version of my files, `git pull origin JA-viewAllPosts-3`
- `Start Tabloid.sln` to open the project in Visual Studio
- If you haven’t created the Tabloid database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
  - Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion/blob/main/SQL/01_Tabloid_Create_DB.sql)
  - ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion/blob/main/SQL/02_Tabloid_Seed_Data.sql)
- Start Tabloid.slc by clicking the green "play" button in Visual Studio. This will open up a Swagger page with API calls.
- Run the program by navigating in a terminal TabloidFullStack > Tabloid Full Stack > client > tabloid and entering the command "npm start"

After running the application in a browser window. You should see the 'Log in' page with a navigation bar at the top of the page. If not already logged out, log out (with the link in the navbar), and 'Log in' with the following credentials.
email: 

Once you're logged in, click on "All posts" in the nav bar.

You will be directed to a "Post List" page . 
 where you can view the existing posts and available info. 



# Checklist:

- I have opened JA-viewAllPosts-3 in my local repository and made sure I have the SQL database.
 I have ran my application in a browser window to see a 'Log in' page
 After logging in the appropriate credentials, selecting User Profiles in the navbar, directs me to a "All Posts" page.
user in the list should
